reconcile/utils/slo_gatekeeper.py

```diff
@@ -0,0 +1,80 @@
+import logging
+from base64 import b64encode
+
+from reconcile.gql_definitions.fragments.saas_slo_document import (
+    SaasSLODocument,
+    SLONamespacesV1,
+)
+from reconcile.utils.secret_reader import SecretReaderBase
+from reconcile.utils.slo_details import PromCredentials, SLODetails
+
+
+class SLOGateKeeper:
+    def __init__(
+        self, slo_documents: list[SaasSLODocument], secret_reader: SecretReaderBase
+    ):
+        self.secret_reader = secret_reader
+        self.slo_details_list = self._create_SLO_details_list(slo_documents)
+
+    def _create_SLO_details_list(
```

Comment:
Pep8 naming convention:

```suggestion
    def _create_slo_details_list(
```